src/main/java/org/fenixedu/academic/domain/student/personaldata/ProfessionalStatusType.java

```diff
@@ -0,0 +1,47 @@
+package org.fenixedu.academic.domain.student.personaldata;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.commons.i18n.LocalizedString;
+
+public class ProfessionalStatusType extends ProfessionalStatusType_Base {
+    protected ProfessionalStatusType() {
+        super();
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    public static ProfessionalStatusType create(String code, LocalizedString name, boolean active) {
+        if (findByCode(code).isPresent()) {
+            throw new DomainException(
+                    BundleUtil.getString(Bundle.APPLICATION, "error.ProfessionalStatusType.code.already.exists", code));
+        }
+
+        ProfessionalStatusType type = new ProfessionalStatusType();
+        type.setCode(code);
+        type.setName(name);
+        type.setActive(active);
+        return type;
+    }
+
+    public void delete() {
+        setRootDomainObject(null);
+        super.deleteDomainObject();
+    }
+
```

Comment:
The `setCode(code)` method should be overriden in order to check if already exists another instance with the same code